app/controllers/application_controller/api_authenticator.rb

```diff
@@ -0,0 +1,39 @@
+class ApplicationController
+  class ApiAuthenticator < Struct.new(:controller)
+    attr_reader :current_user
+
+    def require_authentication
+      set_current_user(raise_errors: true)
+    end
+
+    def api_key
+      controller.request.headers['HTTP_API_KEY']
```

Comment:
I would like to figure out a better way to do this without passing the controller object to the authenticator. Or am I just overthinking this?